OpenRA.Mods.Cnc/Traits/EnergyWall.cs

```diff
@@ -0,0 +1,117 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Will open and be passable for actors that appear friendly when there are no enemies in range.")]
+	public class EnergyWallInfo : BuildingInfo, IConditionConsumerInfo, IRulesetLoaded
+	{
+		[FieldLoader.Require]
+		[WeaponReference]
+		[Desc("The weapon to attack units on top of the wall with when activated.")]
+		public readonly string Weapon = null;
+
+		[ConsumedConditionReference]
+		[Desc("Boolean expression defining the condition to activate this trait.")]
+		public readonly ConditionExpression ActiveCondition = null;
```

Comment:
Just noting for rebasing that `ConditionExpression` should be `BooleanExpression` here.